src/codeocean/data_asset.py

```diff
@@ -331,3 +331,16 @@ def transfer_data_asset(self, data_asset_id: str, transfer_params: TransferDataP
             f"data_assets/{data_asset_id}/transfer",
             json=transfer_params.to_dict()
         )
+
+    def paginate_search(self, data_asset_search_params: DataAssetSearchParams) -> Iterator[DataAsset]:
```

Comment:
can you pull this method up right after the `search_data_assets` for better context?